crates/bls-gadgets/src/y_to_bit.rs

```diff
@@ -37,15 +42,15 @@ impl<P: Bls12Parameters> YToBitGadget<P> {
                 let half = P::Fp::modulus_minus_one_div_two();
                 let y_value = pk.y.get_value().get()?;
                 if y_value.into_repr() > half {
-                    Ok(y_value - &(P::Fp::from_repr(half) + &P::Fp::one()))
+                    Ok(y_value - &P::Fp::from_repr(half))
                 } else {
                     Ok(y_value)
                 }
             } else {
                 Err(SynthesisError::AssignmentMissing)
             }
         })?;
-        let y_bit_lc = y_bit.lc(CS::one(), half_plus_one_neg);
+        let y_bit_lc = y_bit.lc(CS::one(), half_neg);
```

Comment:
Maybe adjust the `enforce smaller than modulus minus one div two` comment below to be "enforce smaller than half" for clarity?